src/Grid/grid.jl

```diff
@@ -820,3 +821,88 @@ struct SurfaceOrientationInfo
     #flipped::Bool
     #shift_index::Int
 end
+
+
+"""
+    InterfaceTransformation
+
+Orientation information for 1D and 2D interfaces in 2D and 3D elements respectively. 
+This information is used to construct the transformation matrix to
+transform the quadrature points from face_a to face_b achieving synced
+spatial coordinates. Face B's orientation relative to Face A's can
+possibly flipped (i.e. the vertices indices order is reversed)
+and the vertices can be rotated against each other. 
+The reference orientation of face B is such that the first node
+has the lowest vertex index. Thus, this structure also stores the
+shift of the lowest vertex index which is used to reorient the face in
+case of flipping ["transform_interface_point!"](@ref).
+Take for example the faces
+```
+2           3
+| \\         | \\
+|  \\        |  \\
+| A \\       | B \\ 
+|    \\      |    \\
+3-----1     1-----2  
+```
+which are rotated against each other by 240° after tranfroming to an
+equilateral triangle (shift index is 2) or the faces
+```
+2           3
+| \\         | \\
+|  \\        |  \\
+| A \\       | B \\ 
+|    \\      |    \\
+1-----3     1-----2  
+```
+which are flipped against each other, note that face B has its reference node shifted by 2 indices
+so the face is tranformed into an equilateral triangle then rotated 120°, flipped about the x axis then
+rotated -120° and tranformed back to the reference triangle.Any combination of these can happen. 
+"""
+struct InterfaceTransformation
+    flipped::ScalarWrapper{Bool}
+    shift_index::ScalarWrapper{Int}
+    lowest_node_shift_index::ScalarWrapper{Int}
+end
```

Comment:
Could we use a mutable struct here?
```suggestion
mutable struct InterfaceTransformation
    flipped::Bool
    shift_index::Int
    lowest_node_shift_index::Int
end
```

Reply:
Didn't notice that, thanks!